### PR TITLE
DBContext is just a Context

### PR DESCRIPTION
**Pull request by zeripath**
This PR removes some of the specialness from the DBContext and makes it context
This allows us to simplify the GetEngine code to wrap around any context in future
and means that we can change our loadRepo(e Engine) functions to simply take contexts.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by lafriks** [comment]
unit tests fail

File: models/attachment.go [hunk 1]
```diff
@@ -221,20 +222,20 @@ func DeleteAttachmentsByComment(commentID int64, remove bool) (int, error) {
...
-	return updateAttachment(db.DefaultContext().Engine(), atta)
+	return updateAttachment(db.GetEngine(db.DefaultContext), atta)
 }
 
 // UpdateAttachmentByUUID Updates attachment via uuid
-func UpdateAttachmentByUUID(ctx *db.Context, attach *Attachment, cols ...string) error {
+func UpdateAttachmentByUUID(ctx context.Context, attach *Attachment, cols ...string) error {
 	if attach.UUID == "" {
-		return fmt.Errorf("Attachement uuid should not blank")
+		return fmt.Errorf("attachment uuid should not blank")
```
Comment (KN4CK3R) [inline models/attachment.go]: Should that be lowercase? Could add a `be`.

File: models/statistic.go [hunk 2]
```diff
@@ -23,20 +23,20 @@ type Statistic struct {
...
+	stats.Counter.PublicKey, _ = db.GetEngine(db.DefaultContext).Count(new(PublicKey))
 	stats.Counter.Repo = CountRepositories(true)
-	stats.Counter.Watch, _ = db.DefaultContext().Engine().Count(new(Watch))
-	stats.Counter.Star, _ = db.DefaultContext().Engine().Count(new(Star))
-	stats.Counter.Action, _ = db.DefaultContext().Engine().Count(new(Action))
-	stats.Counter.Access, _ = db.DefaultContext().Engine().Count(new(Access))
+	stats.Counter.Watch, _ = db.GetEngine(db.DefaultContext).Count(new(Watch))
+	stats.Counter.Star, _ = db.GetEngine(db.DefaultContext).Count(new(Star))
+	stats.Counter.Action, _ = db.GetEngine(db.DefaultContext).Count(new(Action))
+	stats.Counter.Access, _ = db.GetEngine(db.DefaultContext).Count(new(Access))
```
Comment (KN4CK3R) [inline models/statistic.go:31]: Could introduce a variable to reduce the calls.

**Review by KN4CK3R — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
Please resolve the conflicts.

File: models/attachment.go (hunk 1 above)
Comment (zeripath) [reply models/attachment.go]: It should be lowercase - golang errors are not supposed to start with a capital letter: https://github.com/golang/go/wiki/CodeReviewComments#error-string

Although I noted that the verb was missing I'd have to double check that nothing is actually checking for the string as is before adding in the `be`

File: models/statistic.go (hunk 2 above)
Comment (zeripath) [reply models/statistic.go:31]: As you might imagine, the code was a result of a mass find and replace.

Comment (KN4CK3R) [reply models/statistic.go:31]: Sure, but could be changed while working at it.

**Comment by zeripath** [comment]
make lgtm work

**Pull request merged**